packages/react-components/src/components/TextFieldWithMention/TextFieldWithMention.tsx

```diff
@@ -159,8 +159,7 @@ export const TextFieldWithMention = (props: TextFieldWithMentionProps): JSX.Elem
       const mention = htmlStringForMentionSuggestion(suggestion, localeStrings);
 
       // update plain text with the mention html text
-      const newPlainText =
-        inputTextValue.substring(0, currentTriggerStartIndex) + mention + inputTextValue.substring(selectionEnd);
+      const newPlainText = inputTextValue.substring(0, currentTriggerStartIndex);
```

Comment:
It looks like `inputTextValue.substring(selectionEnd);` will be lost in case if mention is added in the middle of the text